pkg/registry/debian/debian.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2024 The OSS Rebuild Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debian
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/google/oss-rebuild/internal/httpx"
+	"github.com/pkg/errors"
+)
+
+var (
+	registryURL         = "https://deb.debian.org/debian"
+	binaryReleaseRegexp = regexp.MustCompile(`(\+b[\d\.]+)$`)
+)
+
+type ControlStanza struct {
+	Fields map[string][]string
+}
+
+type DSC struct {
+	Stanzas []ControlStanza
+}
+
+// Registry is a debian package registry.
+type Registry interface {
+	Artifact(context.Context, string, string, string) (io.ReadCloser, error)
+	DSC(context.Context, string, string, string) (string, *DSC, error)
+}
+
+// HTTPRegistry is a Registry implementation that uses the debian HTTP API.
+type HTTPRegistry struct {
+	Client httpx.BasicClient
+}
+
+func (r HTTPRegistry) get(ctx context.Context, url string) (io.ReadCloser, error) {
+	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	resp, err := r.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, errors.Errorf("fetching artifact: %v", resp.Status)
+	}
+	return resp.Body, nil
+}
+
+func PoolURL(component, name, artifact string) string {
+	prefixDir := name[0:1]
+	// Most packages are in a prefix dir matching their first letter. The exception are lib* files which are in dirs matching lib*.
```

Comment:
nit: Maybe just kill the first sentence. You mainly want to describe the exceptional case: "When packages begin with "lib," the dir prefix excludes that portion of the name."